src/commonControls/questionnaireControl/QuestionnaireControl.ts

```diff
@@ -0,0 +1,1578 @@
+        // if (choiceId) {
+        //     resultBuilder.addAct(
+        //         new QuestionAnsweredAct(this, {
+        //             questionId,
+        //             choiceId,
+        //             userAnsweredWithExplicitValue: false,
+        //             userMentionedQuestion: false,
+        //             renderedChoice: content.choices[choiceIndex].prompt,
+        //             renderedQuestion: content.questions[questionIndex].prompt,
+        //             renderedQuestionShortForm: content.questions[questionIndex].promptShortForm,
+        //         }),
+        //     );
+        // } else {
+        //     resultBuilder.addAct(
+        //         new AnswerClearedAct(this, {
+        //             questionId,
+        //             userMentionedQuestion: false,
+        //             renderedQuestion: content.questions[questionIndex].prompt,
+        //             renderedQuestionShortForm: content.questions[questionIndex].promptShortForm,
+        //         }),
+        //     );
+        // }
+
+        this.clearCompletionFlag(); // user interacted with the and so the questionnaire should stick around
+        this.inputWasAnswerByTouch = true;
+        return;
+    }
+
+    private isCompletionRequestByVoice(input: ControlInput): boolean {
+        try {
+            okIf(InputUtil.isIntent(input, GeneralControlIntent.name));
+            const { feedback, action, target } = unpackGeneralControlIntent(
+                (input.request as IntentRequest).intent,
+            );
+            okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, [$.Feedback.Affirm, $.Feedback.Disaffirm]));
+            okIf(InputUtil.actionIsMatch(action, this.props.interactionModel.actions.complete));
+            return true;
+        } catch (e) {
+            verifyErrorIsGuardFailure(e);
+            return false;
+        }
+    }
+
+    private isCompletionRequestByTouch(input: ControlInput): boolean {
+        try {
+            okIf(InputUtil.isAPLUserEventWithMatchingControlId(input, this.id));
+            const userEvent = input.request as interfaces.alexa.presentation.apl.UserEvent;
+            okIf(userEvent.arguments !== undefined);
+            okIf(userEvent.arguments.length === 2);
+            const kind = (input.request as interfaces.alexa.presentation.apl.UserEvent).arguments![1];
+            okIf(kind === 'complete');
+            return true;
+        } catch (e) {
+            verifyErrorIsGuardFailure(e);
+            return false;
+        }
+    }
+
+    private async handleCompletionRequest(input: ControlInput, resultBuilder: ControlResultBuilder) {
+        const content = this.getQuestionnaireContent(input);
+
+        const validationResult = await evaluateValidationProp(this.props.valid, this.state, input);
+        if (validationResult === true) {
+            this.setCompletionFlag();
```

Comment:
Ok, so you're supporting complete status update only on APL screen but not through voice currently?

Reply:
this one method is common to both APL and voice flow.  Perhaps it would be less confusing to have two methods so that all the names match up (and perhaps a common coreImpl)?